GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTNanopbHelpers.m

```diff
@@ -188,6 +195,76 @@ gdt_cct_IosClientInfo GDTCCTConstructiOSClientInfo() {
   return iOSClientInfo;
 }
 
+gdt_cct_NetworkConnectionInfo GDTCCTConstructNetWorkConnectionInfo() {
+  gdt_cct_NetworkConnectionInfo networkConnectionInfo = gdt_cct_NetworkConnectionInfo_init_default;
+  if (GDTCORPlatformIsIOS()) {
+    networkConnectionInfo.network_type = GDTCCTNetworkConnectonInfoNetworkType();
+    if (networkConnectionInfo.network_type != gdt_cct_NetworkConnectionInfo_NetworkType_NONE) {
+      networkConnectionInfo.has_network_type = 1;
+      if (networkConnectionInfo.network_type == gdt_cct_NetworkConnectionInfo_NetworkType_MOBILE) {
+        networkConnectionInfo.mobile_subtype = GDTCCTNetworkConnectionInfoNetworkMobileSubtype();
+        if (networkConnectionInfo.mobile_subtype !=
+            gdt_cct_NetworkConnectionInfo_MobileSubtype_UNKNOWN_MOBILE_SUBTYPE) {
+          networkConnectionInfo.has_mobile_subtype = 1;
+        }
+      }
+    }
+  } else if (GDTCORPlatformIsMacOS() || GDTCORPlatformIsTvOS() || GDTCORPlatformIsWatchOS()) {
+    SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
+    // Currently set all non-iOS platform network type to WIFI if network is reachable.
+    if (reachabilityFlags & kSCNetworkReachabilityFlagsReachable) {
+      networkConnectionInfo.has_network_type = 1;
+      networkConnectionInfo.network_type = gdt_cct_NetworkConnectionInfo_NetworkType_WIFI;
+    }
+  }
+  return networkConnectionInfo;
+}
+
+gdt_cct_NetworkConnectionInfo_NetworkType GDTCCTNetworkConnectonInfoNetworkType() {
+  gdt_cct_NetworkConnectionInfo_NetworkType networkType =
+      gdt_cct_NetworkConnectionInfo_NetworkType_NONE;
+  SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
+  if (reachabilityFlags & kSCNetworkReachabilityFlagsIsWWAN) {
+    networkType = gdt_cct_NetworkConnectionInfo_NetworkType_MOBILE;
+  } else {
+    networkType = gdt_cct_NetworkConnectionInfo_NetworkType_WIFI;
+  }
+  return networkType;
+}
+
+gdt_cct_NetworkConnectionInfo_MobileSubtype GDTCCTNetworkConnectionInfoNetworkMobileSubtype() {
+  static NSDictionary<NSString *, NSNumber *> *ctRadioAccessTechnologyToNetworkSubType;
+  static dispatch_once_t onceToken = 0;
+  dispatch_once(&onceToken, ^{
+    ctRadioAccessTechnologyToNetworkSubType = @{
+      CTRadioAccessTechnologyGPRS : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_GPRS),
```

Comment:
You'll need to wrap all the CTRadioAccess enums in a custom enum defined differently based on platform. It probably makes sense to write that part in GDTCORPlatform.